### PR TITLE
Apply hinting factor rcParam in all cases.

### PR DESCRIPTION
**Pull request by QuLogic**
This is only explicitly used in Agg, so any other users get the default in the C++ code. Thus if you modify this setting and use some other backend, you get very weird behaviour. Positioning sometimes breaks and glyphs are shrunk or cropped.

@anntzer this is why just changing a random `hinting_factor` produced super-weird results yesterday.

Don't know if it's worth backporting; I guess no-one ever tried modifying this setting or they would have complained. Also, trying to see if a simple test of some sort can be done.

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [?] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
Hmm, something might still be cached somewhere as changing the hinting factor in the middle of a script causes all text to be the wrong width.

**Comment by QuLogic** [comment]
@anntzer True, it just makes it more difficult to write a test.

**Comment by QuLogic** [comment]
I think that should fix the caching issue, and it even fixes the SVG+Agg save from above, without the first change. I guess the first change can still go in for consistency.

But thinking about it, SVG and PDF render text externally, so it may be better to have those use `hinting_factor=1` since that's the most-likely method to be used by external renderers. Maybe can discuss that in a separate issue where we also might decide to turn it off entirely.

**Comment by anntzer** [comment]
Ah, the static was tricky...
I would favor moving the discussion of entirely dropping hinting_factor (which I agree with) to a separate issue, not clear there's much gain of doing so in a piecemeal fashion.

**Comment by QuLogic** [comment]
Not sure why I needed the `clear` and `set_size` as I thought the same stuff is done in the constructor, but it matches what's done in normal rendering code and it works, so...

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -1366,7 +1366,12 @@ def is_opentype_cff_font(filename):
 _fmcache = None
 
 
-get_font = lru_cache(64)(ft2font.FT2Font)
+_get_font = lru_cache(64)(ft2font.FT2Font)
+
+def get_font(filename, hinting_factor=None):
+    if hinting_factor is None:
+        hinting_factor = rcParams['text.hinting_factor']
+    return _get_font(filename, hinting_factor)
```
Comment (Kojoley) [inline lib/matplotlib/font_manager.py]: Is not it equals to?
```python
@lru_cache(64)
def get_font(filename, hinting_factor=None):
    if hinting_factor is None:
        hinting_factor = rcParams['text.hinting_factor']
    return ft2font.FT2Font(filename, hinting_factor)
```

Comment (QuLogic) [reply lib/matplotlib/font_manager.py]: That would cache `hinting_factor=None` with whatever the value of the rcParam was at first call.

**Comment by anntzer** [comment]
thanks

**Pull request merged**